### ISSUE TITLE
TCO: zeta.table.limit is applied before filter

### ISSUE THREAD

**Issue by Maikel-Shinoda** [GIP-SmartMercial/xyna-factory#173]
Adjust the function "getTCOs()" in the class "https://github.com/GIP-SmartMercial/xyna-factory/blob/main/modules/xmcp/guihttp/mdmimpl/TimeControlledOrdersServicesImpl/src/xmcp/factorymanager/impl/TimeControlledOrdersServicesServiceOperationImpl.java" to limit only after filtering etc.

**Comment by Maikel-Shinoda** [comment]
So far:

In BatchProcessSearch.java there is already a filter, sort and limit mechanism. You can pass an object of type "BatchProcessSelectImpl" which may contain some column filters that are applied when searching for mathing batchProcessIds. These are then sorted and filtered.
The problem is, that the method "searchBatchProcessIds()" has no input parameter to determine whether to search for archived TCOs or not - it always adds them to the result. As a consequence, the limiting mechanism is applied on a Set of batchProcessIds which may not be the final result set. Depending on the sorting this can lead to cutting off some desired results.

The calling function getTCOs() in the interface class TimeControlledOrderServicesServiceOperationImpl then applies another filter to get rid of the archived TCOs if not selected.
In my opinion, this filter should be implemented in the BatchProcessSearch (for instance by passing an additional input parameter to the method "searchBatchProcesses()").

Finally, another limiting mechanism is applied to the result set, using the zeta.table.limit.

A quick and dirty fix would probably be to set the limit of the "searchBatchProcesses()" call in TimeControlledOrderServicesServiceOperationImpl.getTCOs() to Integer.MAX_VALUE. This would so to speak disable the limiting mechanism in "searchBatchProcesses()".
This is not my recommended solution though because it would cause the method "searchBatchProcesses()" to retrieve the details for every batch process which fits the filter criteria, neglecting the zeta.table.limit, which can be a huge number of calls for elements that do not even appear in the result set.

Perhaps there is yet another issue because I observed that there were sometimes missing elements in the result set even if I checked the option "show archived TCOs".
